src/domain/formatter/diff-formatter.ts

```diff
@@ -0,0 +1,174 @@
+// src/domain/formatter/diff-formatter.ts
+import chalk from 'chalk';
+import { table } from 'table';
+import { FormatType, FormatterOptions, DirectoryNode, IFormatter } from './types.js';
+import { FormatterUtils, colorMap } from './utils.js';
+import { DiffAnalysis, FileDetails } from '../../types/index.js';
+
+export class DiffFormatter implements IFormatter {
+  constructor(
+    private options: FormatterOptions = {
+      format: FormatType.TREE,
+    }
+  ) {}
+
+  updateOptions(newOptions: Partial<FormatterOptions>) {
+    this.options = { ...this.options, ...newOptions };
+  }
+
+  format(analysis: DiffAnalysis) {
+    switch (this.options.format) {
+      case FormatType.TREE:
+        return this.formatTree(analysis);
+      case FormatType.JSON:
+        return this.formatJson(analysis);
+      default:
+        return this.formatPlain(analysis);
+    }
+  }
+
+  formatDetails(details: FileDetails) {
+    let output = '';
+
+    // Header
+    output += chalk.bold(FormatterUtils.getChangeIcon('file'));
+    output += chalk.gray(`Comparing ${details.fromRef} → ${details.toRef}\n\n`);
```

Comment:
```suggestion
    let output = `${chalk.bold(FormatterUtils.getChangeIcon('file'))}${chalk.gray(`Comparing ${details.fromRef} → ${details.toRef}\n\n`)}`;
```
이 부분은 조건 분기에 따라서 output이 변경되는 부분이 아니니까 이런식으로 초기값을 주고 시작해도 괜찮아 보여요. 

아니면 FormatterUtils, Comparing을 변수화해서 output에 넣어주는 것도 좋을듯?

Reply:
👍 👍 
음... `Header` 컴포넌트를 만들어서 재활용하는 것도 좋아보이는데 1)우선 제안주신 사항으로 반영하고 2)더 좋은 방법이 있을 것 같은데 (예를들면 view만 담당하는 서비스를 만든다던지) 요건 한번 생각해보겠습니다!